SwiftzTests/UserExample.swift

```diff
@@ -10,37 +10,53 @@ import Swiftz
 
 // A user example
 // an example of why we need SYB, Generics or macros
-public class User : JSONDecodable {
+class User {
+	let id : UInt64
 	let name : String
 	let age : Int
 	let tweets : [String]
 	let attr : String
+	let balance : Double
+	let admin : Bool
 
-	public init(_ n : String, _ a : Int, _ t : [String], _ r : String) {
+	init(_ i : UInt64, _ n : String, _ a : Int, _ t : [String], _ r : String, _ b : Double, _ ad : Bool) {
+		id = i
 		name = n
 		age = a
 		tweets = t
 		attr = r
+		balance = b
+		admin = ad
 	}
+}
 
-	// JSON
-	public class func create(x : String) -> Int -> ([String] -> String -> User) {
-		return { y in { z in { User(x, y, z, $0) } } }
-	}
-
-	public class func fromJSON(x : JSONValue) -> User? {
-		let p1 : String? = x <? "name"
-		let p2 : Int? = x <? "age"
-		let p3 : [String]? = x <? "tweets"
-		let p4 : String? = x <? "attrs" <> "one" // A nested keypath
-		return User.create
+extension User : JSONDecodable {
+	class func fromJSON(x : JSONValue) -> User? {
+		let p1 : UInt64? = x <? "id"
+		let p2 : String? = x <? "name"
+		let p3 : Int? = x <? "age"
+		let p4 : [String]? = x <? "tweets"
+		let p5 : String? = x <? "attrs" <> "one" // A nested keypath
+		let p6 : Double? = x <? "balance"
+		let p7 : Bool? = x <? "admin"
```

Comment:
When you go past a few statements without telling the Swift compiler the type it will fail with:
``` Expression was too complex to be solved in reasonable time; consider breaking up the expression into distinct sub-expressions
``` since it would have to infer the types.